catalog/plugin/catalog-plugin-security-audit/src/main/java/org/codice/alliance/catalog/plugin/security/audit/SecurityAuditPlugin.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.alliance.catalog.plugin.security.audit;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.collections.MapUtils;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.util.impl.Requests;
+import ddf.security.common.audit.SecurityLogger;
+
+public class SecurityAuditPlugin implements AccessPlugin {
+
+    private List<String> auditAttributes = new ArrayList<>();
+
+    @Override
+    public CreateRequest processPreCreate(CreateRequest createRequest)
+            throws StopProcessingException {
+        return createRequest;
+    }
+
+    @Override
+    public UpdateRequest processPreUpdate(UpdateRequest updateRequest,
+            Map<String, Metacard> existingMetacards) throws StopProcessingException {
+        if (updateRequest != null && Requests.isLocal(updateRequest)) {
```

Comment:
Could we change this if to do the opposite check and then do a quick return to avoid so much nesting. Could also add the `MapUtils.isNotEmpty(existingMetacards)` check to this if and remove two levels of nesting.